code/game/machinery/machinery.dm

```diff
@@ -351,3 +351,37 @@ Class Procs:
 	icon_state = "autolathe"
 	density = TRUE
 	anchored = TRUE
+
+//fuel pomp
+/obj/structure/machinery/fuelpump
+	name = "\improper Fuel Pump"
+	layer = ABOVE_MOB_LAYER
+	desc = "It is a machine that pump fuel in lifeboats."
+	icon = 'icons/obj/structures/machinery/fuelpump.dmi'
+	icon_state = "fuelpump_off"
+	density = TRUE
+	anchored = TRUE
+
+
+/obj/structure/machinery/fuelpump/update_icon()
+	if(stat & NOPOWER)
+		icon_state = "fuelpump_off"
```

Comment:
you can probably put return here and remove the else clause completely

Reply:
thanks for taking the time to help. i am going for a version without switch to all the different stage.